### ISSUE TITLE
Bug: sort-interfaces changes overload ordering for constructors

### ISSUE THREAD

**Issue by plneple** [azat-io/eslint-plugin-perfectionist#414]
### Describe the bug

`sort-interfaces`  changes overload ordering of constructor in interfaces, which can cause existing code to break.

Method overloads do not change order, so constructors should not change too.

### Code example

```
interface Interface {
  new (value: number | string): number;
  new (value: number): unknown;
}
```

### ESLint version

latest

### ESLint Plugin Perfectionist version

latest

### Additional comments

I have written the following test to verify the issue in this repo.
The case with `new` clause fails, while other cases pass.

```js
ruleTester.run(`${ruleName}(${type}): sorts interface overloads`, rule, {
  invalid: [
    {
      errors: [
        {
          data: {
            right: 'method',
            left: 'methodA',
          },
          messageId: 'unexpectedInterfacePropertiesOrder',
        },
      ],
      output: dedent`
        interface Interface {
          method(value: number | string): number;
          method(value: number): unknown;

          methodA(value: number | string): number;
          methodA(value: number): unknown;
        }
      `,
      code: dedent`
        interface Interface {
          methodA(value: number | string): number;
          methodA(value: number): unknown;

          method(value: number | string): number;
          method(value: number): unknown;
        }
      `,
      options: [options],
    },
  ],
  valid: [
    {
       code: dedent`
         interface Interface {
           method(value: number | string): number;
           method(value: number): unknown;
         }
       `,
       options: [options],
     },
     {
       code: dedent`
         interface Interface {
           new (value: number | string): number;
           new (value: number): unknown;
         }
       `,
       options: [options],
     },
   ],
})
```

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @plneple,

Thank you for the reproduction cases and the tests! The fix will be available in next release.

**Comment by azat-io** [comment]
Fixed in b4ee1cce0176ab00446bf2d06cb5caa7fd8b28a4.
Released in [v4.4.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.4.0).
